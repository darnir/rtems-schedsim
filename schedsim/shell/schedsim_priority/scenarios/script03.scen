echo "*** TEST 03 ***"
rtems_init
echo "=== Create 0x0a010001 ==="
task_create TA1 1
echo "** Mode - query"
task_mode
echo "** Mode - set no timeslicing"
task_mode -t
echo "** Mode - set timeslicing"
task_mode -T
echo "** Mode - set no timeslicing and no preempt"
task_mode -t -p
echo "** Mode - set timeslicing and preempt"
task_mode -T -P
echo "** Mode - query"
task_mode
echo "*** END OF TEST 03 ***"
exit
# We will not get here
