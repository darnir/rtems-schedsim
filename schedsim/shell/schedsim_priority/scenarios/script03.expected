*** TEST 03 ***
  CPU 0: Thread Heir: 0x09010001 priority=255
  CPU 0: Thread Executing: 0x09010001 priority=255
=== Create 0x0a010001 ===
Creating task with:  Preemption: yes Timeslicing: no
Task (TA1) created: id=0x0a010001, priority=1
Task (TA1) starting: id=0x0a010001, priority=1
  CPU 0: Thread Heir: 0x0a010001 priority=1
  CPU 0: Thread Executing: 0x0a010001 priority=1
** Mode - query
Previous Mode: Preemption: yes Timeslicing: no
** Mode - set no timeslicing
Previous Mode: Preemption: yes Timeslicing: no
Current Mode:  Preemption: yes Timeslicing: no
** Mode - set timeslicing
Previous Mode: Preemption: yes Timeslicing: no
Current Mode:  Preemption: yes Timeslicing: yes
** Mode - set no timeslicing and no preempt
Previous Mode: Preemption: yes Timeslicing: yes
Current Mode:  Preemption: no Timeslicing: no
** Mode - set timeslicing and preempt
Previous Mode: Preemption: no Timeslicing: no
Current Mode:  Preemption: yes Timeslicing: yes
** Mode - query
Previous Mode: Preemption: yes Timeslicing: yes
*** END OF TEST 03 ***
